Manage.c

```diff
@@ -537,7 +535,7 @@ static void add_detached(struct mddev_dev *dv, int fd, char disp)
 		sfd = dev_open(buf, O_RDONLY);
 		if (sfd >= 0) {
 			/* Not detached */
-			close(sfd);
+			close_fd(&sfd);
```

Comment:
Actually, in this case we should stay with `close()` because the routine `add_detached()` must keep fd verification so it is unnecessarily doubled. Immediately after `close()` `continue;` is done so reassignment doesn't matter and it is accessed always after dev_open()- must be reinitialized at this point.

Reply:
fixed